### ISSUE TITLE
update-git-for-windows seems to be broken on 2.45.2.vfs.0.0

### ISSUE THREAD

**Issue by tiagomacarios** [microsoft/git#655]
z:\o2\src>git --version
git version 2.45.2.vfs.0.0
 
z:\o2\src>git update-git-for-windows -y
curl: (43) A libcurl function was given a bad argument
 
z:\o2\src>git --version
git version 2.45.0.vfs.0.0
 
z:\o2\src>git update-git-for-windows -y
Git 2.45.0.vfs.0.0 (64-bit)
Up to date

**Comment by dscho** [comment]
This seems to be caused by a bug in cURL where it does not handle the `--write-out` option correctly:

```console
$ sh -x $(which git-update-git-for-windows)
+ use_recently_seen=no
+ test --test-version-compare '!=' ''
+ update_git_for_windows
[...]
+ echo 'Update 2.45.2.windows.1 is available'
Update 2.45.2.windows.1 is available
++ http_get https://api.github.com/repos/git-for-windows/git/releases/latest
++ url=https://api.github.com/repos/git-for-windows/git/releases/latest
+++ mktemp -t gfw-httpget-XXXXXXXX.txt
++ output=/tmp/gfw-httpget-ruMLqjGG.txt
+++ curl --silent --show-error --output /tmp/gfw-httpget-ruMLqjGG.txt --write-out '%{http_code}' https://api.github.com/repos/git-for-windows/git/releases/latest
curl: (43) A libcurl function was given a bad argument
++ code=000
++ return 43
+ releases=
+ return
```

If I remove the `--write-out '%{http_code}'` option, the command succeeds (but of course the HTTP code is no longer reported).

https://github.com/curl/curl/issues/13845 looks closely related, and I am currently building cURL v8.8.0 with https://github.com/curl/curl/commit/9aa1d412b814a40868558da51a6ab28ce1384a58 applied on top, to see whether this fixes the bug.

**Comment by dscho** [comment]
> I am currently building cURL v8.8.0 with [curl/curl@9aa1d41](https://github.com/curl/curl/commit/9aa1d412b814a40868558da51a6ab28ce1384a58) applied on top, to see whether this fixes the bug.

Sadly, it does not address the bug.

**Comment by dscho** [comment]
> > I am currently building cURL v8.8.0 with [curl/curl@9aa1d41](https://github.com/curl/curl/commit/9aa1d412b814a40868558da51a6ab28ce1384a58) applied on top, to see whether this fixes the bug.
> 
> Sadly, it does not address the bug.

Aha! I managed to test with the original DLL the entire time because I failed to realize that the custom build I ran in place would pick up `/mingw64/bin/libcurl-4.dll` instead. Once I overwrite that with the just-built binary, it _does_ fix the bug!

**Comment by derrickstolee** [comment]
Reopening until new release bits can be generated.

**Comment by dscho** [comment]
> Reopening until new release bits can be generated.

They are [generated](https://github.com/microsoft/git/actions/runs/9402980517) as we speak ;-)

**Comment by derrickstolee** [comment]
```
NORTHAMERICA+dstolee@stoleedesk2023 MINGW64 ~
$ git update-microsoft-git
curl: (43) A libcurl function was given a bad argument

NORTHAMERICA+dstolee@stoleedesk2023 MINGW64 ~
$ git version
git version 2.45.2.vfs.0.1
```

Doesn't seem to have stuck.

**Comment by dscho** [comment]
> Doesn't seem to have stuck.

Indeed. I had to [kicked off another `sync` run in `git-sdk-64`](https://github.com/git-for-windows/git-sdk-64/actions/runs/9403501634) that [updated curl to the -2 version](https://github.com/git-for-windows/git-sdk-64/commit/ca2d56e17d40a8b3152c6f6f0e6f295688d16331), and then I re-ran [the `windows_pkg` job](https://github.com/microsoft/git/actions/runs/9402980517/job/25900373783). The release workflow run is now completing, and I will test it immediately once it's done.

**Comment by dscho** [comment]
Now, that's better:

![image](https://github.com/microsoft/git/assets/127790/351ad03f-21c3-4299-bd30-2e6fc7814e3d)

There's room for improvement (Portable Git should decline to run `update-git-for-windows`, for example), but the bug is gone.

**Comment by derrickstolee** [comment]
I confirmed this worked for me, too. In my opinion, this issue can be closed when 2.45.2.vfs.0.1 is available as a pre-release, to match the exposure of 2.45.2.vfs.0.0.

**Comment by dscho** [comment]
[Published](https://github.com/microsoft/git/releases/tag/v2.45.2.vfs.0.1)!
